scripts/rpmbuildinfo

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+
+#
+# Copyright (c) 2016  Bernhard M. Wiedemann <bernhard@zq1.de>
+# Copyright (c) 2016  Dennis Gilmore <dennis@ausil.us>
+# Copyright (c) 2016  Simon "HW42" Geiser <hw42@ipsumj.de>
+# Copyright (c) 2016  Wojciech Porczyk <woju@invisiblethingslab.com>
+# Copyright (c) 2019  Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>
+# Copyright (c) 2021  Frédéric Pierret <frederic.pierret@qubes-os.org>
+#
+# All rights reserved.
+# 
+# Permission to use, copy, modify, and distribute this software for any purpose
+# with or without fee is hereby granted, provided that the above copyright
+# notice and this permission notice appear in all copies.
+# 
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT OF THIRD PARTY RIGHTS. IN
+# NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+# OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
+# OR OTHER DEALINGS IN THE SOFTWARE.
+# 
+# Except as contained in this notice, the name of a copyright holder shall not
+# be used in advertising or otherwise to promote the sale, use or other dealings
+# in this Software without prior written authorization of the copyright holder.
+#
+
+set -e -o pipefail
+
+getos() {
+    # shellcheck disable=SC1091
+    test -r /etc/os-release && . /etc/os-release
+    if test -z "${ID}"; then
+        ID="$(cat /etc/system-release)"
+    fi
+    printf '%s' "${ID}"
+}
+
+RPM_BUILD_ROOT="$1"
+RPM_BUILD_DIR="$2"
+RPM_PACKAGE_EPOCH="${3:-0}"
+RPM_PACKAGE_NAME="$4"
+RPM_PACKAGE_VERSION="$5"
+RPM_PACKAGE_RELEASE="$6"
+RPM_ARCH="$7"
+
+RPM_BUILD_NAME="${RPM_PACKAGE_NAME}-${RPM_PACKAGE_VERSION}-${RPM_PACKAGE_RELEASE}.${RPM_ARCH}"
+RPM_BUILD_ARCH="$(uname -m)"
```

Comment:
This will get you the host arch, not the build arch.

Reply:
It depends on definition. Looking at https://manpages.debian.org/testing/dpkg-dev/deb-buildinfo.5.en.html, this perhaps tries to be Build-Architecture counterpart which means *distro* architecture, but that could differ from `uname -m`. It seems to me this was added from your request to track the build arch in case of noarch packages, and seems reasonable to me. But it's not really possible to discuss the correctness of a value without a definition of what that value is supposed to mean in the first place.